test/extended/dr/common.go

```diff
@@ -303,3 +312,236 @@ func checkSSH(node *corev1.Node) {
 func ssh(cmd string, node *corev1.Node) (*e2essh.Result, error) {
 	return e2essh.IssueSSHCommandWithResult(cmd, e2e.TestContext.Provider, node)
 }
+
+// InstallSSHKeyOnControlPlaneNodes will create a new private/public ssh keypair,
+// create a new secret for both in the openshift-etcd-operator namespace. Then it
+// will append the public key on the host core user authorized_keys file with a daemon set.
+func InstallSSHKeyOnControlPlaneNodes(oc *exutil.CLI) error {
+	const name = "dr-ssh"
+	const namespace = "openshift-etcd"
+
+	err := createPrivatePublicSSHKeySecret(oc, name, namespace)
+	if err != nil {
+		return err
+	}
+
+	err = createKeyInstallerDaemon(oc, name, namespace)
+	if err != nil {
+		return err
+	}
+
+	err = ensureControlPlaneSSHAccess(oc, name, namespace)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// ensureControlPlaneSSHAccess will test that the private key generated and installed with installSSHKeyOnControlPlaneNodes
+// is working on all control plane nodes. This effectively polls until the pod executing ssh succeeds reaching all nodes.
+func ensureControlPlaneSSHAccess(oc *exutil.CLI, name string, namespace string) error {
+	const sshPath = "/tmp/ssh"
+	const containerName = "ssh-key-tester"
+
+	var cpNodeInternalIps []string
+	for _, node := range masterNodes(oc) {
+		framework.Logf("CP Node meta: %s, status: %s", node.ObjectMeta.String(), node.Status.String())
+		for _, address := range node.Status.Addresses {
+			if address.Type == corev1.NodeInternalIP {
+				cpNodeInternalIps = append(cpNodeInternalIps, address.Address)
+			}
+		}
+	}
+
+	framework.Logf("found internal IPs: %v", cpNodeInternalIps)
+
+	testScript := fmt.Sprintf(`
+		#!/bin/bash
+		CORE_SSH_BASE_DIR=$HOME/.ssh
+		SSH_MOUNT_DIR=%s
+		P_KEY=$SSH_MOUNT_DIR/privKey
+		# we can't change the permissions on the secret mount, thus we copy it to HOME
+		mkdir -p $CORE_SSH_BASE_DIR && chmod 700 $CORE_SSH_BASE_DIR
+		cp $P_KEY $CORE_SSH_BASE_DIR/id_rsa
+		P_KEY=$CORE_SSH_BASE_DIR/id_rsa
+		chmod 600 $P_KEY
+
+		ls -l $P_KEY
+		
+		NODE_IPS=( %s )
+		for i in "${NODE_IPS[@]}"; do
+		  echo "testing SSH to [$i]"
+		  until ssh -i $P_KEY -o StrictHostKeyChecking=no -q core@${i} exit
+		  do
+			echo "Waiting for ssh connection on ${i}"
+			sleep 5
+		  done 
+		  echo ""
+		  echo "SSH to [$i] was successful!"
+		done
+			`, sshPath, strings.Join(cpNodeInternalIps, " "))
+
+	podSpec := applycorev1.PodSpec().WithHostNetwork(true).WithRestartPolicy(corev1.RestartPolicyOnFailure)
+	podSpec.Containers = []applycorev1.ContainerApplyConfiguration{
+		*applycorev1.Container().
+			WithName(containerName).
+			WithSecurityContext(applycorev1.SecurityContext().WithPrivileged(true)).
+			// TODO(thomas): is there a different rhel based image with ssh available?
+			WithImage("registry.redhat.io/rhel7/rhel-tools").
```

Comment:
is this test intended to be part of the conformance suite?

see: https://github.com/openshift/origin/blob/master/test/extended/util/image/README.md

Reply:
Nope. if you have an offline available image that has ssh in it... I'm more than happy to add it.

Reply:
thanks, then let me see if that image works properly.

Reply:
works like a charm. thanks @bparees!